### ISSUE TITLE
define ⊇ analogous to ⊆

### ISSUE THREAD

**Issue by StefanKarpinski** [JuliaLang/julia#21461]
Discovered that this operator is undefined by default, since ⊆ is, it should also be defined. We should probably take a pass through all defined operators that have a naturally symmetric pair and make sure that they have the appropriate definitions.

**Comment by bramtayl** [comment]
Does it make sense to define ⊇ if you can get the same thing with ⊆ and a reverse order? Blah blah blah properly factored functions blah blah blah

**Comment by StefanKarpinski** [comment]
As much as it makes sense to have `>` even though you could write everything with `<`.

**Comment by TotalVerb** [comment]
Isn't there an issue for lowering things like `EXPR1 ⊇ EXPR2` to `let x = EXPR1, y = EXPR2; y ⊆ x; end`; or perhaps I'm misremembering?

**Comment by bramtayl** [comment]
> As much as it makes sense to have > even though you could write everything with <

Does it?

**Comment by StefanKarpinski** [comment]
> Isn't there an issue for lowering things like `EXPR1 ⊇ EXPR2` to `let x = EXPR1, y = EXPR2; y ⊆ x; end`; or perhaps I'm misremembering?

This would be an alternative general solution.

**Comment by StefanKarpinski** [comment]
> Does it?

Are you proposing that we get rid of the `>` operator?

**Comment by bramtayl** [comment]
Well it's something I've pondered, anyway. At one point I tried going through the names in Base to see if I could refactor them better and it was a thought I had.

**Comment by StefanKarpinski** [comment]
Not being able to write `a > b` would be really annoying. It would be better to do the lowering thing where `a > b` is lowered to the moral equivalent of `b < a`.

**Comment by SaschaMann** [comment]
I wrote the definition for `⊇, ⊉, ⊋` before your comments, so might as well put it here even if the general solution suggested above seems nicer than having
```julia
⊇(l, r) = issubset(r, l)
⊉(l::Set, r::Set) = !⊇(l, r)
⊋(l::Set, r::Set) = <(r, l)
```
for each operator with a mirrored counterpart.

If it helps, I can put up the WIP PR in a bit, once the tests finish running.

**Comment by JeffBezanson** [comment]
You can't do this with lowering if you want to have both `<` and `>` available as first-class functions. We already have the definition `>(x, y) = y < x`, along with a note that new types should implement `<`.

This isn't a matter of proper code factoring, just notational convenience.

**Comment by StefanKarpinski** [comment]
One could always make `>` a shorthand for `(args...)->(<)(reverse(args)...)` everywhere – then it would be possible to use it as a first-class object as well, but then you might as well just provide that as a default definition and let people provide a new definition of the want.

**Comment by dpsanders** [comment]
That definition wouldn't have a splatting penalty?

**Comment by bramtayl** [comment]
> notational convenience

I'm not too invested here, but I can't think of an example where `>` is more convenient than `<`

**Comment by TotalVerb** [comment]
`pop!(A) > pop!(A)`?

**Comment by bramtayl** [comment]
Clever. Though I'm not sure that's the clearest way to write that code. 

```
last_one = pop!(A)
second_to_last_one = pop!(A)
second_to_last_one < last_one
```

**Comment by JeffBezanson** [comment]
At risk of taking this question too seriously, `sort(x, lt = <)` vs. `sort(x, lt = >)` ?

**Comment by bramtayl** [comment]
`sort(x, lt = <, rev = true)`?

**Comment by bramtayl** [comment]
Or even:

```
reverse(f) = (a, b) -> f(b, a)
sort(x, lt = reverse(<)
```

**Comment by JeffBezanson** [comment]
And perhaps you get tired of writing `reverse(<)`, and want a short name for it. If only there were a standard symbol for that.

**Comment by martinholters** [comment]
> If only there were a standard symbol for that.

`⇄(<)` ?

**Comment by StefanKarpinski** [comment]
Obviously everything can be expressed with just `<`. Similarly, we can express all control flow constructs with gotos. Minimalism is not the end goal, just a sometimes-useful heuristic. People like to use both `<` and `>` in mathematical notation, and we're interested in making it easy to just transcribe mathematics in a way that matches the original text – excluding `>` does not help that.

**Comment by JeffreySarnoff** [comment]
Julia enables and eases communication of computational ideas among humans;  imo, pair accordingly.

**Comment by JeffBezanson** [comment]
Closed by #21469
